src/pygeoprocessing/kernels.py

```diff
@@ -0,0 +1,295 @@
+            of the kernel will have values of ``0.0``.
+        pixel_radius: The radius of the target kernel, in pixels.  If ``None``,
+            then ``math.ceil(max_distance)`` will be used.
+        normalize: Whether to normalize the kernel.
+
+    Returns:
+        ``None``
+    """
+    def _linear_decay(dist):
+        return (max_distance - dist) / max_distance
+
+    create_kernel(
+        target_kernel_path=target_kernel_path,
+        function="(max_dist - dist) / max_dist",
+        max_distance=max_distance,
+        pixel_radius=pixel_radius,
+        normalize=normalize
+    )
+
+
+def normal_distribution_kernel(
+        target_kernel_path: Text,
+        sigma: Union[int, float],
+        n_std_dev: Union[int, float] = 3,
+        pixel_radius: Union[int, float] = None,
+        normalize: bool = True):
+    """Create an decay kernel following a normal distribution.
+
+    Args:
+        target_kernel_path: The path to where the kernel will be written.
+            Must have a file extension of ``.tif``.
+        sigma: The width (in pixels) of a standard deviation.
+        n_std_dev: The number of standard deviations to include in the kernel.
+            The kernel will have values of 0 when at a distance of
+            ``(sigma * n_std_dev)`` away from the centerpoint.
+        pixel_radius: The radius of the target kernel, in pixels.  If ``None``,
+            then ``math.ceil(sigma * n_std_dev)`` will be used.
+        normalize: Whether to normalize the kernel.
+
+    Returns:
+        ``None``
+    """
+    def _normal_decay(dist):
+        return (1 / (2 * numpy.pi * sigma ** 2)) * numpy.exp(
+            (-dist ** 2) / (2 * sigma ** 2))
+
+    create_kernel(
+        target_kernel_path=target_kernel_path,
+        function=_normal_decay,
+        max_distance=(sigma * n_std_dev),
+        pixel_radius=pixel_radius,
+        normalize=normalize
+    )
+
+
+def create_kernel(
+        target_kernel_path: str,
+        function: Union[str, Callable],
+        max_distance: Union[int, float],
+        pixel_radius=None,
+        normalize=True):
+    """
+    Create a kernel raster based on pixel distance from the centerpoint.
+
+    Args:
+        target_kernel_path (string): The path to where the target kernel should
+            be written on disk.  If this file does not have the suffix
+            ``.tif``, it will be added to the filepath.
+        function (callable): A python callable that takes as input a
+            1D numpy array and returns a 1D numpy array.  The input array will
+            contain float32 distances to the centerpoint pixel of the kernel.
+        max_distance (float): The maximum distance of kernel values from
+            the center point.  Values outside of this distance will be set to
+            ``0.0``.
+        pixel_radius=None (float): The radius (in pixels) of the target kernel.
+            If ``None``, the radius will be ``math.ceil(max_distance)``.
```

Comment:
I agree, `apothem` is much more accurate (even if not commonly recognized), particularly since each pixel more accurately represents a square of points.  With that model, we don't even need the +0.5 pixel adjustment!

Reply:
I've renamed `pixel_radius` to `apothem` in 4e99b96